### PR TITLE
[TASK] Merge EXT:solrfluidgrouping into EXT:solr

### PR DESCRIPTION
**Pull request by sfroemkenjw**
# What this pr does

Migrate EXT:solrfluidgrouping into EXT:solr

Fixes: #3631

---

- [x] Squash this on merge

### CODE REVIEW

**Review by bmack — APPROVED** [review]
LGTM

**Comment by dkd-friedrich** [comment]
I've pushed some adjustments and bugfixes, but there is still an issue with "allowGetParameterSwitch" and parameter `tx_solr[grouping]`, grouping cannot be disabled as intended. I'll analyze this next week

File: Classes/Query/Modifier/Grouping.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+use ApacheSolrForTypo3\Solr\Domain\Search\SearchRequestAware;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Modifies a query to add grouping parameters
+ *
+ * @author Ingo Renner <ingo@typo3.org>
+ * @author Frans Saris <frans@beech.it>
+ */
+class Grouping implements Modifier, SearchRequestAware
```
Comment (dkd-kaehm) [inline Classes/Query/Modifier/Grouping.php:33]: Whhoops, already overlapped:

See: https://github.com/TYPO3-Solr/ext-solr/blob/092e2a729471607b2b23a7519d2ebdeb14f579b3/Documentation/Releases/solr-release-12-0.rst?plain=1#L35

**Review by dkd-kaehm — CHANGES_REQUESTED** [review]
We want rebase against main.
To many changes overlapped with movements to PSRs

**Review by bmack — APPROVED** [review]

**Review by bmack — APPROVED** [review]

**Pull request merged**